stdlib/public/Platform/Darwin.swift

```diff
@@ -14,3 +14,18 @@
 
 public let MAP_FAILED =
   UnsafeMutableRawPointer(bitPattern: -1)! as UnsafeMutableRawPointer!
+
+@available(swift, deprecated: 3.0, message: "Please use 'Double.pi' or '.pi' to get the value of correct type and avoid casting.")
+public let M_PI = Double.pi
+
+@available(swift, deprecated: 3.0, message: "Please use 'Double.pi / 2' or '.pi / 2' to get the value of correct type and avoid casting.")
+public let M_PI_2 = Double.pi / 2
+
+@available(swift, deprecated: 3.0, message: "Please use 'Double.pi / 4' or '.pi / 4' to get the value of correct type and avoid casting.")
+public let M_PI_4 = Double.pi / 4
+
+@available(swift, deprecated: 3.0, message: "Please use '2.squareRoot()'.")
+public let M_SQRT2 = 2.0.squareRoot()
```

Comment:
I suppose we could implement some silly intrinsic like `Builtin.isStaticallyTrue(self >= 0)` and then do something like this:

```swift
if Builtin.isStaticallyTrue(self >= 0) {
  return Builtin.sqrt_fp64(self.rawValue)
}
return sqrt(self)
```

(or however we spell intrinsics)